### PR TITLE
[Doc] Release notes 11.0.0

### PR DESCRIPTION
**Pull request by jcalcaben**
## Description

Update the changelog with release notes for the 11.0.0 release.

Also updates the compatibility table.

## Related Issue

Closes PWA-1875

## Acceptance

@davemacaulay 
@tkacheva 

### Verification Stakeholders

Any core developer

@davemacaulay 
@tkacheva 

## Test Plan

1. Open CHANGELOG.md
2. Verify content is accurate
3. Open magento-compatibility.js
4. Verify content is accurate

#### Test scenario(s) for direct fix/feature
Example
-   Verify user is able to apply filters on category page
-   Verify user is able to apply filters on search page

#### Test scenario(s) for any existing impacted features/areas
Example
-   Verify user is able to sort data after applying filters on category page
-   Verify user is able to sort data after applying filters on search page

#### Test scenario(s) for any Magento Backend Supported Configurations
Example
- Update default Sort value in backend and repeat above scenarios

#### Is Browser/Device testing needed?
Example
- Yes, browser testing is needed as X UI component may be impacted on <browser>
- Yes, device testing is needed as X UI component may be impacted on <mobile|desktop|etc>

#### Any ad-hoc/edge case scenarios that need to be considered?
Example
- Apply all filters to get 0 results and then remove filters to see respective products 

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Comment by revanth0212** [comment]
Shouldn't this be against the release branch?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,335 +1,348 @@
...
+[#3246]: https://github.com/magento/pwa-studio/pull/3246
+### Cypress tests
 
-![Custom scaffolding template](https://user-images.githubusercontent.com/4692281/112341120-25f5a680-8c8f-11eb-9174-122dcc1eb5c4.png)
+[Cypress](https://www.cypress.io/) is an end-to-end testing suite written in JavaScript.
+This release adds this framework to the PWA Studio project to increase testing automation and reduce the time spent on manual testing.
+This will enable the team to release new versions faster and more often.
 
-### MegaMenu component
+Integration tests for PageBuilder and the new Wish List feature are included in this release.
```
Comment (davemacaulay) [inline CHANGELOG.md:73]: Could we add a link to where these tests live within the PWA Studio repository?

We would want our customers to be able to utilize these tests in their own CI/CD pipelines eventually, so might be worth adding a note around that.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,335 +1,348 @@
...
+[#3253]: https://github.com/magento/pwa-studio/pull/3253
+[#3195]: https://github.com/magento/pwa-studio/pull/3195
 
-### Store switcher
+### Virtual Product types
 
-The Venia UI library now provides components that support multiple store views.
-These components let customers switch between the different store views defined in the Magento backend.
+This release contains initial work to support Virtual Product types.
+The current implementation only lets you browse and view Virtual Product types in your storefront.
```
Comment (davemacaulay) [inline CHANGELOG.md:100]: I think we should make it clearer around the issues with the work we've done. The fact they cannot be purchased and will display a notice on the product page sounds like something helpful in these notes.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,335 +1,348 @@
...
-This release improves the performance of the Venia storefront and its underlying components.
-These improvements include:
+### Extensible payment methods
 
-- Enabling text compression for the UPWARD-JS server
-- Removing unused JavaScript in Venia
-- Investigating and improving render blocking and response in Venia
+Two new extensions points for the Venia UI package have been added in this release.
+The `editablePaymentTypes` target lets you add new editable payment methods to your storefronts, and
+the `summaryPagePaymentTypes` target lets you add a custom payment summary in the checkout summary page.
```
Comment (davemacaulay) [inline CHANGELOG.md:113]: Could we directly link both of these targets to their associated documentation?

**Review by davemacaulay — CHANGES_REQUESTED** [review]
Overall looks great, a few minor updates :)

**Comment by jcalcaben** [comment]
@revanth0212 good catch! I changed the base to release/11.0

**Review by davemacaulay — APPROVED** [review]
